ambry-store/src/main/java/com.github.ambry.store/Log.java

```diff
@@ -153,19 +201,25 @@ LogSegment getNextSegment(LogSegment segment) {
   }
 
   /**
-   * @param name the name of the segment required.
-   * @return a {@link LogSegment} with {@code name} if it exists.
+   * Returns the {@link LogSegment} that is logically before the given {@code segment}.
+   * @param segment the {@link LogSegment} whose "previous" segment is required.
+   * @return the {@link LogSegment} that is logically before the given {@code segment}.
    */
-  LogSegment getSegment(String name) {
-    return segmentsByName.get(name);
+  LogSegment getPrevSegment(LogSegment segment) {
```

Comment:
The index is expected to have an atomic view of the store whereas the log may not. This is because log segments are added to the store during the commit phase of compaction but removed during the cleanup. There is an extended period of time where the log's data structures contain both the old and new log segments. On the other hand, the index always has an atomic view - it contains either the old or the new index segments and is better equipped to answer questions about capacity and log segment counts. Index segments have always determined the order of log segments (and this ordering comes from `Offset`).